src/kudu/consensus/raft_consensus.cc

```diff
@@ -579,6 +589,12 @@ Status RaftConsensus::StartElection(ElectionMode mode, ElectionContext context)
     LockGuard l(lock_);
     RETURN_NOT_OK(CheckRunningUnlocked());
 
+    if (!persistent_vars_->is_start_election_allowed()) {
```

Comment:
where are we disabling the failure detector by this variable?

Reply:
We are _not_ disabling the failure detector. As Mike and Vinay have suggested in previous comments, we went with the route of not starting elections to handle all edge/race conditions that would have risen had we tried stopping the failure detector instead.

So the flow would be this: if you want to disable elections, you'd set a sys var on MySQL that would call `SetAllowStartElection(false)` which would persist this setting to disk. Whenever the peer attempts to start an election, it will check the value of `allow_start_election` and bail out if `False`.